File: tests/conformance/programs/iterator.ts
function* countUp(n: number): Iterable<number> {
    let i = 0;
    while (i < n) {
        yield i;
        i = i + 1;
    }
}
function main(): void {
    let total = 0;
    for (const v of countUp(5)) {
        total = total + v;
    }
    console.log(total);
    for (const v of countUp(3)) {
        console.log(v);
    }
}
main();

File: tests/conformance/programs/iterator.cs
using System;

static class Program
{
    static System.Collections.Generic.IEnumerable<int> countUp(int n)
    {
        var i = 0;
        while (i < n)
        {
            yield return i;
            i = i + 1;
        }
    }
    static void main()
    {
        var total = 0;
        foreach (var v in countUp(5))
        {
            total = total + v;
        }
        Console.WriteLine(total);
        foreach (var v in countUp(3))
        {
            Console.WriteLine(v);
        }
    }
    
    static void Main() { main(); }
}
